### PR TITLE
fix(mock-data): Fix failing functional tests.

### PR DESCRIPTION
**Pull request by nimishamukherjee**
Fix failing functional tests. Add new work item to the top of the list.

### CODE REVIEW

**Comment by fabric8cd** [comment]
@nimishamukherjee snapshot fabric8-ui is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-1990-fabric8-planner.int.rdu2c.fabric8.io

**Comment by sanbornsen** [comment]
@nimishamukherjee check my comments here - https://github.com/fabric8io/fabric8-planner/pull/1960#pullrequestreview-43301425

**Comment by nimishamukherjee** [comment]
@sanbornsen shouldn't the mock be fixed?
When a work item gets added or edited we need to
1. Add the work item to the list
2. Display the work item matches only if it matches the the applied filters
The function loadWorkItems does this. Why do we need to replicate the steps under loadWorkItems under add and edit?
We need to update the mock so that it mimics adding of the new work item to the top of the list.

**Comment by fabric8cd** [comment]
@nimishamukherjee snapshot fabric8-ui is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-1990-fabric8-planner.int.rdu2c.fabric8.io

**Comment by fabric8cd** [comment]
@nimishamukherjee snapshot fabric8-ui is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-1990-fabric8-planner.int.rdu2c.fabric8.io

File: src/app/components/planner-list/planner-list.component.ts [hunk 1]
```diff
@@ -461,6 +461,17 @@ export class PlannerListComponent implements OnInit, AfterViewInit, DoCheck, OnD
           }
         })
     );
+this.eventListeners.push(
```
Comment (sanbornsen) [inline src/app/components/planner-list/planner-list.component.ts]: Why do we need these two for? Aren't next two doing the same job?

File: src/app/components/planner-list/planner-list.component.ts [hunk 2]
```diff
@@ -496,13 +515,34 @@ export class PlannerListComponent implements OnInit, AfterViewInit, DoCheck, OnD
...
-        this.loadWorkItems();
+      this.workItemService.editWIObservable.subscribe(updatedItem => {
+        let index = this.workItems.findIndex((item) => item.id === updatedItem.id);
+        if(this.filterService.doesMatchCurrentFilter(updatedItem)){
+          console.log('Updated WI matches the applied filters')
+          if (index > -1) {
+            this.workItems[index] = updatedItem;
+          } else {
+            //add the WI at the top of the list
+            this.workItems.splice(0, 0, updatedItem);
```
Comment (sanbornsen) [inline src/app/components/planner-list/planner-list.component.ts]: Why would this be in top of the list? It should stay in the same index position in the list.

File: src/app/services/filter.service.ts [hunk 3]
```diff
@@ -88,12 +88,20 @@ export class FilterService {
    * @returns Boolean
    */
   doesMatchCurrentFilter(workItem): Boolean {
+    let activeFilters = cloneDeep(this.activeFilters);
+    //Remove the parentexists filter
+    let index = activeFilters.findIndex(item => {
+      return item.paramKey == 'filter[parentexists]'
+    })
+    if(index > -1 ) {
+      activeFilters.splice(index,1);
```
Comment (sanbornsen) [inline src/app/services/filter.service.ts]: Please give a space after ','  :P

Comment (nimishamukherjee) [inline src/app/services/filter.service.ts]: Space is there  - it is (index, 1)

**Pull request merged**